infrastructure/code/autofix.go

```diff
@@ -41,29 +40,48 @@ func (a AutofixUnifiedDiffSuggestion) GetUnifiedDiffForFile(filePath string) str
 	return a.UnifiedDiffsPerFile[filePath]
 }
 
-func (s *SnykCodeHTTPClient) GetAutofixDiffs(ctx context.Context, baseDir string, options AutofixOptions) (unifiedDiffSuggestions []AutofixUnifiedDiffSuggestion, err error) {
+func (s *SnykCodeHTTPClient) GetAutofixDiffs(ctx context.Context, baseDir string, options AutofixOptions) (
+	unifiedDiffSuggestions []AutofixUnifiedDiffSuggestion,
+	status AutofixStatus,
+	err error,
+) {
 	method := "GetAutofixDiffs"
 	logger := config.CurrentConfig().Logger().With().Str("method", method).Logger()
 	span := s.instrumentor.StartSpan(ctx, method)
 	defer s.instrumentor.Finish(span)
 
-	var response AutofixResponse
 	requestId, err := performance2.GetTraceId(ctx)
 	if err != nil {
 		logger.Err(err).Msg(failedToObtainRequestIdString + err.Error())
-		return unifiedDiffSuggestions, err
+		return nil, failed, err
 	}
 
 	logger.Info().Str("requestId", requestId).Msg("Started obtaining autofix diffs")
 	defer logger.Info().Str("requestId", requestId).Msg("Finished obtaining autofix diffs")
 
-	response, err = s.RunAutofix(span.Context(), options)
-	if err != nil || response.Status == failed.message {
-		logger.Err(err).Msg("error getting autofix suggestions")
-		return unifiedDiffSuggestions, err
+	response, err := s.RunAutofix(span.Context(), options)
+	if err != nil {
+		return nil, failed, err
+	}
+
+	logger.Debug().Msgf("Status: %s", response.Status)
+
+	if response.Status == failed.message {
+		logger.Error().Str("responseStatus", response.Status).Msg("autofix failed")
```

Comment:
This probably should be an info level or warn level, as it's not a technical error, but a business logic outcome, right?

Reply:
Good point, Antoine in our team is currently working on providing more granular error propagation on the backend side. I took this from another place in the codebase. I keep it as ERROR for now and we will revisit this in the downstream services once we add granular errors to the backend and sast-autofix-api.